src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -317,15 +317,18 @@ def refreshCache(self):
             potentialLocations = set()
             potentialLocations.update(possibleLocations)
 
-            # now check for sites in drain and adjust the possible locations
-            # also check if there is at least one site left to run the job
+            # check if there is at least one site left to run the job
             if len(possibleLocations) == 0:
                 newJob['fileLocations'] = loadedJob.get('fileLocations', [])
                 newJob['siteWhitelist'] = loadedJob.get('siteWhitelist', [])
                 newJob['siteBlacklist'] = loadedJob.get('siteBlacklist', [])
                 badJobs[71101].append(newJob)
                 continue
-            else:
+
+            # if agent is in speed drain and has hit the threshold to submit to all sites, we can skip the logic to exclude sites
+            if not self.reqAuxDB.getWMAgentConfig(self.config.Agent.hostName).get("SpeedDrainConfig")['EnableAllSites']['Enabled']:
```

Comment:
how about we break this line down to 2 such that it's more readable?
We also need to check whether `if self.reqAuxDB:` before we call getWMAgentConfig